active-directory-wpf-msgraph-v2/MainWindow.xaml.cs

```diff
@@ -37,9 +42,20 @@ private async void CallGraphButton_Click(object sender, RoutedEventArgs e)
             ResultText.Text = string.Empty;
             TokenInfoText.Text = string.Empty;
 
-            var accounts = await app.GetAccountsAsync();
-            var firstAccount = accounts.FirstOrDefault();
+            IAccount firstAccount;
 
+            // WAM will always get an account in the cache. So if we want
+            // to have a chance to select the accounts interactively, we need to
+            // force the non-account
+            if (howToSignIn.SelectedIndex == 1)
+            {
+                firstAccount = null;
+            }
+            else
+            {
+                var accounts = await app.GetAccountsAsync();
+                firstAccount = accounts.FirstOrDefault();
+            }
```

Comment:
We should have a discussion about prompts as well. In WAM, there are only 2 prompts - "default" and "force". "Force" is very similar to our "select_account". 

The current logic I implemented for mapping is as follows: 

https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/blob/882c815c5e319b5ab356cca1bbfc39ac36dd022b/src/client/Microsoft.Identity.Client/Platforms/Features/WamBroker/WamBroker.cs#L143


- if the user configures `select_account` or `force_login`, wam prompt becomes `force`
- if the user configures `no_prompt` or does not make any configuration, WAM prompt becomes `default`
- all other prompts cannot be used

Happy to re-discuss this point. I think MSAL should also stop defaulting to "select_account" and instead it should not configure any prompt.

Reply:
@bgavrilMS : did you have an opportunity to discuss `Account.WindowsSession` with the WAM people? (or `IPublicClientApplication.WindowsSession`)